playlistmanager.lua

```diff
@@ -635,8 +635,29 @@ function playlist(force_dir)
   if playlist_visible then showplaylist() end
 end
 
+local function check_playlist_savapath()
+  if settings.playlist_savepath == nil or utils.readdir(settings.playlist_savepath) == nil then
+		
+    local default_playlist_savepath = mp.command_native({"expand-path", "~~home/"}).."/playlists"
+    msg.info("The custom playlist save path is invalid. Use the default path: "..default_playlist_savepath)
+		
+    if mp.find_config_file("playlists") then
+      settings.playlist_savepath=mp.find_config_file("playlists")
+    else
+      msg.info("Default playlist save path does not exist. It will be created automatically.")
+      if os.getenv("windir") then
+        os.execute("mkdir "..default_playlist_savepath:gsub('/', '\\'))
+      else
+        os.execute("mkdir "..default_playlist_savepath)
+      end
+      settings.playlist_savepath = default_playlist_savepath
+    end
+  end
+end
+
 --saves the current playlist into a m3u file
 function save_playlist()
+  check_playlist_savapath()
```

Comment:
Can you move this call to after the early return. Also typo `check_playlist_savepath`.

Reply:
> Perhaps the default setting savepath could just be `nil`?
It could. But I think setting a value explicitly is not easy to misunderstand.